### PR TITLE
Fix #294 - Added form validation in front end using JavaScript

### LINKED ISSUE #294: Add error info to specific field and include details
When a user enters data that is not correct entered and hits "Upload" a general error message appears. Please make this easier to understand and add more information.

1. Highlight the box that was erroneous in red and add info that it is erroneous
2. Keep the information that was previously entered in all fields
3. Provide a helper text how the information should be correctly formatted

screenshot from 2017-10-22 17-42-54

See screenshot above. The user does not know what is the correct format. There is no info anywhere what the correct format is like.

### PR DESCRIPTION
**Pull request by djmgit**
Fixes issue #294, Added JS form validation, input filters and
input specific errors.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by vaibhavsingh97** [comment]
@djmgit Please fix conflicts 😄

**Comment by djmgit** [comment]
@vaibhavsingh97 will fix conflicts ASAP

**Comment by djmgit** [comment]
@vaibhavsingh97  resolved conflicts, please review.

File: app/static/js/form.js [hunk 1]
```diff
@@ -34,4 +34,56 @@ $(document).ready(
...
+  if (csv !== undefined && csv !== null && csv !== "") {
+    var csvLines = csv.split("\n");
+    var countLines = 0;
+    csvLines.forEach(function(csvLine) {
+      var line = csvLine.split(",");
+      if (line.length === 4) {
+        countLines += 1;
+      }
+    });
+    if (countLines !== csvLines.length) {
```
Comment (jajodiaraghav) [inline app/static/js/form.js]: Can you please explain why we are counting individual lines. I think `csvLines` will always be equal to `countLines`

Comment (djmgit) [reply app/static/js/form.js]: @jajodiaraghav it checks whether all the lines are of the format _,_,_,_,
This same logic is implemented in the server side counterpart, I have just ported it to JS.

Comment (jajodiaraghav) [reply app/static/js/form.js]: Okay. Right! Thanks 

Looks good!

**Comment by djmgit** [comment]
merging this :)

**Pull request merged**